### PR TITLE
Remove v1alpha4 jobs for CAPZ

### PR DESCRIPTION
**Pull request by Jont828**
Remove v1alpha4 jobs for CAPZ as those tests are no longer needed. Discussed offline with @CecileRobertMichon

### CODE REVIEW

File: config/jobs/kubernetes-sigs/cluster-api-provider-azure/cluster-api-provider-azure-presubmits-main.yaml [hunk 1]
```diff
@@ -494,6 +494,7 @@ presubmits:
     annotations:
       testgrid-dashboards: sig-cluster-lifecycle-cluster-api-provider-azure
       testgrid-tab-name: capz-pr-apiversion-upgrade-main
+      # TODO: If we're removing the v1alpha4 jobs, do we still want to keep the api version upgrade job?
```
Comment (CecileRobertMichon) [inline config/jobs/kubernetes-sigs/cluster-api-provider-azure/cluster-api-provider-azure-presubmits-main.yaml:497]: yes I believe we want to keep that so we don't break upgrade for existing clusters with new versions

Comment (CecileRobertMichon) [reply config/jobs/kubernetes-sigs/cluster-api-provider-azure/cluster-api-provider-azure-presubmits-main.yaml:497]: @Jont828 can you please remove the TODO?

**Review by mboersma — APPROVED** [review]
/lgtm
/approve

The only mentions of `v1alpha4` that remain are in the comments about the "API Version Upgrade" job which we want to leave intact.

**Pull request merged**